src/coreclr/inc/gcinfodecoder.h

```diff
@@ -265,6 +265,7 @@ class BitStreamReader
 
         m_pCurrent = m_pBuffer = dac_cast<PTR_size_t>((size_t)dac_cast<TADDR>(pBuffer) & ~((size_t)sizeof(size_t)-1));
         m_RelPos = m_InitialRelPos = (int)((size_t)dac_cast<TADDR>(pBuffer) % sizeof(size_t)) * 8/*BITS_PER_BYTE*/;
+        m_current = *m_pCurrent >> m_RelPos;
```

Comment:
What guarantees that the bitstream is long enough? Can this hit access violation by reading unmapped memory?

(It seems to be pre-existing issue, but this change makes it more apparent.)

Reply:
This can be fixed by aligning the m_pCurrent to size_t, and ignoring the extra bits introduced by the alignment. It will fix the correctness issue and it may make the bit reader faster - misaligned reads are more expensive in general.

Reply:
Ah, ignore this. I see that the code a few lines above does exactly that.